internal/config/experimental.go

```diff
@@ -3,7 +3,6 @@ package config
 // ExperimentalConfig allows for experimental features to be enabled
 // and disabled.
 type ExperimentalConfig struct {
-	FilesystemStorage ExperimentalFlag `json:"filesystem_storage,omitempty" mapstructure:"filesystem_storage"`
```

Comment:
i just realized that if we remove this check then anyone who has storage configured (ie git/object) but has this field set to `false` (disabling it) will now all of a sudden have it re-enabled on upgrade (since we are skipping the check).

perhaps we need to be a bit smarter about this and warn that the field is deprecated, but only enable it if it is either missing or set to false?

Reply:
Perhaps all we need to do is, as you say, put a warning if it is used, that it is now ignored.
All this does currently is enable / disable parsing of the configuration it self.

The defaults for the configuration that are parsed are to basically fallback to pre-fs backend behaviour, i.e. use a database.

So, this would only be problematic if someone had set the storage type to e.g. local, git or object but hadn't enabled this feature flag. That would cause Flipt to start using the new backends unexpectedly.
They would have made a chosen effort to configure those backends in the first place though.